src/java.base/share/classes/jdk/internal/util/JDKUTF.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2024, Alibaba Group Holding Limited. All Rights Reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package jdk.internal.util;
+
+import jdk.internal.vm.annotation.ForceInline;
+
+/**
+ * Helper to JDK UTF putChar and Calculate length
+ *
+ * @since 24
+ */
+public abstract class JDKUTF {
+    @ForceInline
+    public static int putChar(byte[] buf, int offset, char c) {
```

Comment:
Do you think we can just move the loop here, like:
```java
public static int putChars(byte[] buf, int offset, String str, int strStart, int strEnd) {
    for (int i = strStart; i < strEnd; i++) {
        offset = putChar(buf, offset, str.charAt(i);
    }
}
```
We can even unroll putChar in the loop.

Reply:
This does not work in ObjectOutputStream#writeMoreUTF